scripts/core/editor3/helpers/htmlComesFromDraftjsEditor.ts

```diff
@@ -1,2 +1,18 @@
-export const htmlComesFromDraftjsEditor = (html: string) => new DOMParser().parseFromString(html, 'text/html')
-    .body.querySelector('[data-offset-key]') != null;
+export const htmlComesFromDraftjsEditor = (html: string) => {
+    const parser = new DOMParser().parseFromString(html, 'text/html');
+    const comesFromeditor = parser.querySelector('[data-offset-key]') != null;
+
+    if (comesFromeditor) {
+        return true;
+    }
+
+    // Dragging a single line in chrome removes DraftJS's data
+    // and only sends a <span> containing the dragged text next
+    // to a <meta> tag
```

Comment:
How do you know then if it's a meta+span from superdesk or some text from wikipedia for example? Wouldn't dragging from anywhere produce meta+span ?

Reply:
well I assume if you only drag text, then the embed shouldn't be created, it should just drop that text there, which is what is doing right now

Reply:
It sounds like you can't determine where the text is coming from and make the assumptions based on the content. I call this lying in the code :)

You can make assumptions, but you need naming to reflect that. If it was called `htmlMaybeComesFromDraftjsEditor` it would be more appropriate, because it indicates to the developer that the function is imprecise.

The problem with the current code is that someone may use the function in the future and it will produce false positives. It's best to avoid false positives / false negatives entirely. For example, in this case, if there is no way to determine if the text is coming from draft-js or not, you could rename the function(as it was before your change) to `htmlContainsDraftJsBlock` and it would then be accurate to its name and wouldn't produce false positives/negatives.

Somewhere in the embed/pasting code, you could then add some code doing the guesswork based on content which would produce the same end result without making a function behave differently than its name implies.

Reply:
I can get the old function back and make a new one to "guess" it's just plain text and check both when dropping to determine if the `embed` will be created 👍